packages/components/multiselect/src/Multiselect.tsx

```diff
@@ -0,0 +1,276 @@
+import { Button, IconButton } from '@contentful/f36-button';
+import { TextInput, type TextInputProps } from '@contentful/f36-forms';
+import { CloseIcon, ChevronDownIcon, SearchIcon } from '@contentful/f36-icons';
+import { SkeletonContainer, SkeletonBodyText } from '@contentful/f36-skeleton';
+import { Popover } from '@contentful/f36-popover';
+
+import { getMultiselectStyles } from './Multiselect.styles';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface MultiselectProps
+  extends CommonProps,
+    Pick<
+      TextInputProps,
+      | 'isDisabled'
+      | 'isInvalid'
+      | 'isReadOnly'
+      | 'isRequired'
+      | 'id'
+      | 'defaultValue'
+    > {
+  /**
+   * Set a custom icon for the text input
+   */
+  startIcon?: React.ReactElement;
+
+  children?: React.ReactNode;
+  /**
+   * Placeholder shown before selecting any elements. Defaults to 'Select one or more items'
+   */
+  placeholder?: string;
+
+  currentSelection?: Array<string>;
+
+  /**
+   * If this is set to `true` the text input will be cleared after an item is selected
+   * @default false
+   */
+  clearAfterSelect?: boolean;
+
+  /**
+   * If this is set to `true` there will be a search input inside the drawer
+   * @default false
+   */
+  hasSearch?: boolean;
+
+  /**
+   * Function called whenever the search input value changes
+   */
+  onSearchValueChange?: (event: React.ChangeEvent) => void;
+
+  /**
+   * This is the value will be passed to the `placeholder` prop of the input.
+   * @default "Search"
+   */
+  searchPlaceholder?: string;
+
+  hasNoMatches?: boolean;
+
+  /**
+   * A message that will be shown when it is not possible to find any option that matches the input value
+   * @default "No matches"
+   */
+  noMatchesMessage?: string;
+
+  /**
+   * Use this prop to get a ref to the input element of the component
+   */
+  inputRef?: React.Ref<HTMLInputElement>;
+  /**
+   * Use this prop to get a ref to the toggle button of the component
+   */
+  toggleRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * Use this prop to get a ref to the list of items of the component
+   */
+  listRef?: React.Ref<HTMLUListElement>;
+
+  /**
+   * It sets the width of the list
+   * @default "auto"
+   */
+  listWidth?: 'auto' | 'full';
+
+  /**
+   * It sets the max-height, in pixels, of the list
+   * The default value is the height of 5 single line items
+   * @default 180
+   */
+  listMaxHeight?: number;
+  /**
+   * Sets the list to show its loading state
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * Boolean to control whether or not to render the suggestions box in a React Portal.
+   * Rendering content inside a Portal allows the suggestions box to escape the bounds
+   * of its parent while still being positioned correctly.
+   * Defaults to `false`
+   */
+  usePortal?: boolean;
+}
+
+function _Multiselect(props: MultiselectProps, ref: React.Ref<HTMLDivElement>) {
```

Comment:
```suggestion
function _MultiSelect(props: MultiSelectProps, ref: React.Ref<HTMLDivElement>) {
```

This should be `MultiSelect` in this file as well for the file naming and the documentation.